pkg/network/pod/annotations/generator_test.go

```diff
@@ -135,6 +136,39 @@ var _ = Describe("Annotations Generator", func() {
 			Expect(annotations).To(HaveKeyWithValue(networkv1.NetworkAttachmentAnnot, expectedValue))
 		})
 	})
+
+	Context("Istio annotations", func() {
+		It("should generate Istio annotation when VMI is connected to pod network using masquerade binding", func() {
+			annotationsGenerator := annotations.NewGenerator(clusterConfig)
+
+			vmi := libvmi.New(
+				libvmi.WithInterface(*v1.DefaultMasqueradeNetworkInterface()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			)
+
+			annotations, err := annotationsGenerator.Generate(vmi)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(annotations).To(HaveKeyWithValue(istio.KubeVirtTrafficAnnotation, "k6t-eth0"))
+		})
+
+		DescribeTable("should not generate Istio annotation", func(vmi *v1.VirtualMachineInstance) {
+			annotationsGenerator := annotations.NewGenerator(clusterConfig)
+
+			annotations, err := annotationsGenerator.Generate(vmi)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(annotations).To(Not(HaveKey(istio.KubeVirtTrafficAnnotation)))
+		},
+			Entry("when VMI is not connected to pod network", libvmi.New()),
+			Entry("when VMI is connected to pod network and not using masquerade binding",
```

Comment:
An unrelated topic for a follow up discussion.

The means that for passt, we do not add a special annotation.
And I think ISTIO still works, so I would be interested to get to the bottom of this: Why ISTIO needs it and how it related to Kubevirt. I suspect ISTIO adds some nftables rules to integrate with the ones we add to the masquerade binding, but I am unsure.

@maiqueb , @qinqon , FYI